File: EPSSystem/EPCSystemFrontend/epcsystemfrontend/lib/devicedto.dart
class DeviceDto {
  final int userId;
  final String deviceName;
  final String location;

  DeviceDto({
    required this.userId,
    required this.deviceName,
    required this.location,
  });

  // Add a method to convert to JSON
  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'deviceName': deviceName,
      'location': location,
    };
  }
}

File: EPSSystem/EPCSystemAPI/EPCSystemAPI/Models/DeviceDto.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace EPCSystemAPI.models
{
    public class DeviceDto
    {
        public int UserId { get; set; }
        public string DeviceName { get; set; }
        public string PowerType { get; set; }
        public string DeviceType { get; set; }
        public string Location { get; set; }
    }
}